radixdlt-core/radixdlt/src/main/java/com/radixdlt/consensus/BFTHeader.java

```diff
@@ -116,12 +113,17 @@ public BFTHeader(
 		this.ledgerHeader = ledgerHeader;
 	}
 
+	@JsonCreator
+	public static BFTHeader create(
+		@JsonProperty(value = "view", required = true) long number,
+		@JsonProperty(value = "vertex_id", required = true) HashCode vertexId,
+		@JsonProperty(value = "ledger_header", required = true) LedgerHeader ledgerHeader
+	) {
+		return new BFTHeader(View.of(number), requireNonNull(vertexId), requireNonNull(ledgerHeader));

```

Comment:
I would move validation to constructor as it is public thus still possible to create an invalid Instance.

Reply:
done